core/src/smartcontracts/isi/account.rs

```diff
@@ -578,7 +576,13 @@ pub mod query {
 
             iroha_logger::trace!(%id);
             Ok(Box::new(
-                state_ro.world().domain(id)?.accounts.values().cloned(),
+                state_ro
+                    .world()
+                    .domain(id)?
+                    .accounts
+                    .iter()
+                    .filter_map(|a_id| state_ro.world().account(a_id).ok())
+                    .cloned(),
```

Comment:
I think instead of storing account ids inside domain we can take advantage of the fact that accounts are stored in order so that all accounts belong to the same domain are adjacent. 

This would be similar to how it's done in `account_roles_iter`.

Reply:
Oh interesting, I had not thought of that.